server/autotest_server/__init__.py

```diff
@@ -396,6 +396,8 @@ def update_test_settings(user, settings_id, test_settings, file_url):
             default_env = os.path.join(TEST_SCRIPT_DIR, DEFAULT_ENV_DIR)
             if not os.path.isdir(default_env):
                 subprocess.run([sys.executable, "-m", "venv", default_env], check=True)
+                requirements_path = os.path.join(os.path.dirname(__file__), "../requirements.txt")
```

Comment:
It's overkill to install all of the server dependencies here. In general I don't think it's a good idea to try to import `..config` from the haskell tester setup.

We can discuss later the best way to pass a "default" resolver as a setting, but for now I want you to keep the default as a constant (in the haskell tester setup file) and don't add it to the environment settings. See also my comment below about the schema.